fixed all files and add oop pattern

restuwahyu13 opened the pull request.

restuwahyu13 committed:
update config typescript for better development

restuwahyu13 committed:
fixed package.json for developemtn

add ts-node-dev like nodemon and fixed package structure

restuwahyu13 committed:
fixed file

restuwahyu13 committed:
fixed file

restuwahyu13 committed:
fixed file

restuwahyu13 committed:
fixed files

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
add oop pattern

restuwahyu13 committed:
update file

restuwahyu13 committed:
add opp pattern

restuwahyu13 committed:
Update profileControllers.ts

munirapp commented:
Thanks atas PR-nya om @restuwahyu13, sebelumnya minta maaf saya hold dulu review dan mergenya karena mau saya buatin CI-nya dulu.

restuwahyu13 commented:
oke siap

munirapp commented:
Ada confilict om @restuwahyu13 mungkin bisa diperbaiki dulu

restuwahyu13 commented:
oke siap gan

restuwahyu13 committed:
Merge branch 'master' into master

restuwahyu13 committed:
This package has been deprecated

This is a stub types definition. knex provides its own type definitions, so you do not need this installed.

restuwahyu13 committed:
remove private moditifier

restuwahyu13 committed:
remove private moditifier

restuwahyu13 committed:
change class function name

restuwahyu13 committed:
change class function name

munirapp reviewed (commented):
@types/morgan ini package untuk apa?

restuwahyu13 committed:
add resolve json module

restuwahyu13 committed:
fix typo module name

restuwahyu13 committed:
fix typo call variable

restuwahyu13 commented:
morgan (logger http)

munirapp commented on src/routes/index.ts:
```diff
@@ -1,11 +1,12 @@
 import { Router } from 'express'
 import { Request as Req, Response as Res } from 'expres'
-import profileControllers from '../controllers/profileControllers'
+import { Controller } from '../controllers/root.controller'
```
Ada baiknya root controller dirubah ke index.ts saja, jadinya nanti import-nya tinggal 
import {Controller} from "../controllers/"

munirapp commented on src/controllers/profileControllers.ts:
```diff
@@ -1,8 +1,12 @@
-import profile from "../models/profile";
-import response from "../helper/response";
-import { redisGetAsync, redisSetAsync } from "../../redis";
+import { Model } from '../models/root.model'
```
Ini juga sama catatannya dengan root controller rubah ke index.ts saja supaya lebih ringkas importnya

restuwahyu13 replied:
oke gan, sekalian sama prefix pathnya biar lebih rapih pake module tambahan

munirapp commented on src/models/profile.ts:
```diff
@@ -1,47 +1,44 @@
 }
 
-export default new Profile(knex, "profile");
+const ProfileSchema = new Profile(knex, 'profile')
```
Ini urgensinya apa harus di passing dulu ke dalam variable baru?

restuwahyu13 replied:
biar lebih spesifik aja si jadi antara nama variable di schema sama yang nanti dipanggil namanya match, kalau pake default kan nanti namanya jadi bebas

restuwahyu13 committed:
add logger http morgan

munirapp commented:
Banyak catatan ini harus dikerjakan @restuwahyu13 beberapa sudah saya buatin note. Salah satu yang krusial coba pastikan file konfigurasi docker dan unit test udah sama dengan branch master karena saya lihat ini dua kali failed test di CI

restuwahyu13 commented:
di local si lancar gan, iya nih tapi teryata di docker nya error, laptop saya gk support docker soalnya jadi saya gk bisa test pake docker

munirapp reviewed (commented):
Kalau cuman untuk logger http hilangkan saja, karena projectnya sudah pake nodemon otomatis sudah log httpnya. Belum lagi di real case orang build production menggunakan pm2 jadi kayaknya terlalu overkill kalau ditambahkan lagi http logger

munirapp commented:
> di local si lancar gan, iya nih tapi teryata di docker nya error, laptop saya gk support docker soalnya jadi saya gk bisa test pake docker

Ya udah klo misalnya laptopnya belum support docker gak papa yang penting pastikan sama dulu file2 berikut:

```
Dockerfile
docker-compose.yml
.github/workflows/main.yml
```

restuwahyu13 commented:
> > di local si lancar gan, iya nih tapi teryata di docker nya error, laptop saya gk support docker soalnya jadi saya gk bisa test pake docker
> 
> Ya udah klo misalnya laptopnya belum support docker gak papa yang penting pastikan sama dulu file2 berikut:
> 
> ```
> Dockerfile
> docker-compose.yml
> .github/workflows/main.yml
> ```

restuwahyu13 commented:
siap gan

munirapp commented:
Saya cek memang kayaknya file dockernya gak update dengan repo yang terbaru, saya sarankan PR ini di close aja terus buat PR yang baru dengan branch baru, jadi PR yang lama masih tetap terkeep di branch yang lama. Jangan lupa pastikan source master hasil **fork** sama dengan source master **induk**

munirapp commented:
Btw, silahkan bertanya om kalau dirasa ada yang kurang jelas atau kurang sesuai. Kita sama2 diskusi

restuwahyu13 commented:
oke siap saya close, biar enak via telegram saja mas diskusnya tele saya @restuvh705

munirapp commented:
Oke kalau bgtu nanti aya invite di grup open source
